File: problems/p/patuljci/patuljci.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    numbers_seven_dwarves = 100
    dwarves = list()
    for _ in range(9):
        dwarves.append(int(input()))

    total_dwarves = sum(dwarves)
    difference_dwarves = total_dwarves - numbers_seven_dwarves

    found_legit = False
    for i in range(9):
        value_1 = dwarves[i]
        for j in range(9):
            if i == j:
                continue
            value_2 = dwarves[j]
            if value_1 + value_2 == difference_dwarves:
                found_legit = True
                dwarves.remove(value_1)
                dwarves.remove(value_2)
                break

        if found_legit:
            break

    for dwarf in dwarves:
        print(dwarf)


if __name__ == '__main__':
    main()

File: problems/p/patuljci/patuljci.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

#define NUMBERS_SEVEN_DWARVES 100
#define NUMBER_DWARVES 9

int main() {
    int dwarves[9];
    int total_dwarves, difference_dwarves, dwarf_number_1, dwarf_number_2;
    bool found_legit;

    total_dwarves = 0;
    for (int i=0; i<NUMBER_DWARVES; i++) {
        cin >> dwarves[i];
        total_dwarves += dwarves[i];
    }

    difference_dwarves = total_dwarves - NUMBERS_SEVEN_DWARVES;

    found_legit = false;

    dwarf_number_1 = 0;
    dwarf_number_2 = 0;
    
    for (int i=0; i<NUMBER_DWARVES; i++){
        dwarf_number_1 = dwarves[i];
        for (int j=0; j<NUMBER_DWARVES; j++) {
            if (i == j) {
                continue;
            }
            dwarf_number_2 = dwarves[j];

            if (dwarf_number_1 + dwarf_number_2 == difference_dwarves) {
                found_legit = true;
                break;
            }
        }
        if (found_legit) {
            break;
        }
    }

    for (int i=0; i<NUMBER_DWARVES; i++){
        if (dwarves[i] == dwarf_number_1 || dwarves[i] == dwarf_number_2) {
            continue;
        }
        cout << dwarves[i] << endl;
    }
}